deployment/roles/common/tasks/repo.yml

```diff
@@ -1,34 +1,21 @@
 ---
-- name: Create archives of the monorepository
-  shell: |
-    (git ls-tree -r HEAD --name-only | sed '/^contracts$/d') | xargs tar zcfv monorepo.tar.gz --files-from -
-    cd contracts; (git ls-tree -r HEAD --name-only) | xargs tar zcfv ../contracts.tar.gz --files-from -
-  delegate_to: 127.0.0.1
-  become: false
-  args:
-    chdir: "{{ lookup('env', 'PWD') }}/.."
+- name: Get bridge repo
+  git:
+    repo: "{{ bridge_repo }}"
+    dest: "{{ bridge_path }}"
+    force: no
+    update: no
+    version: "master"
 
-- name: Copy the archives
-  copy:
-    src: ../../../../{{ item }}
-    dest: "{{ bridge_path }}/"
-  with_items:
-    - monorepo.tar.gz
-    - contracts.tar.gz
-
-- name: Untar the archives
+- name: Checkout proper branch
   shell: |
-    tar zxfv monorepo.tar.gz && rm monorepo.tar.gz
-    mkdir -p contracts && tar zxfv contracts.tar.gz -C ./contracts && rm contracts.tar.gz
+    # If this is a forked PR, the branch needs to be fetched in a special way
+    [ ! "CIRCLE_PR_NUMBER" = "" ] && git fetch --force origin -- "{{ bridge_repo_branch }}/head:remotes/origin/{{ bridge_repo_branch }}"
```

Comment:
`CIRCLE_PR_NUMBER` is only available on forked PR. So if it is available, we fetch the forked branch in a special way.